### ISSUE TITLE
[query] Make it easier to `ground` sets of attribute idents

### ISSUE THREAD

**Issue by ncalexan** [mozilla/mentat#705]
Right now, `ground` (see #99 and friends) doesn't make it easy to query over multiple attributes, which I believe is one of its principle uses in Datomic.  E.g., for
```
[:find
 ...
 :where
  [(ground [:credential/username :credential/password]) [?a ...]]
  [?e ?a _ ?tx]
]
```

I get
```
CC known empty: Type mismatch: var(?a) can't be ValueTypeSet({Ref}), because it's already ValueTypeSet({Keyword}).
```
That's correct, but not as helpful as it could be.  This ticket tracks trying to make `ground` bind its variable later, or otherwise not be as strict about its value type.  (Note that if the order of where constraints is reversed, this is fine -- we appear to be correctly interpreting the ident keyword as a ref.)

**Comment by rnewman** [comment]
We might consider doing this during the existing pattern walk that Thom added; it’s apparent from the first walk that `?a` is a ref here.

The broader question is whether we want to support dualism between keywords and refs:

```edn
[:find …
 :where
 [(ground :foo/bar ?dual)]
 [?e ?dual ?v]                      ; Dual must be an entity.
 [?x :attribute/keyword ?dual]]     ; Dual must be a keyword.
```

Right now we don’t support that: ground it twice!

In your case I think you can type-hint:

```edn
[:find …
 [(ref ?a)]
 [(ground […] [?a ...])]
 [?e ?a _ ?tx]]
```

**Comment by rnewman** [comment]
Fixed by #713.
